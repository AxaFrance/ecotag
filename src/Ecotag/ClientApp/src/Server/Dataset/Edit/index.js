export {default} from './Edit.container.js';
