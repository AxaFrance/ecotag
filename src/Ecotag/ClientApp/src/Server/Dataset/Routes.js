import React from 'react';
import {Route, Switch, useRouteMatch} from 'react-router-dom';
import Confirm from './Add/Confirm';
import New from './Add/New';
import List from './List';
import Edit from './Edit/index.js';

const Routes = () => {
    const {url} = useRouteMatch();
    return (
        <Switch>
            <Route exact path={`${url}`} component={List}/>
            <Route exact path={`${url}/new`} component={New}/>
            <Route exact path={`${url}/confirm`} component={Confirm}/>
            <Route exact path={`${url}/:id`} component={Edit}/>
        </Switch>
    );
};

export default Routes;
