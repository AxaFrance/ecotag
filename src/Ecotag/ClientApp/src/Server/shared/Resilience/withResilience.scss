.resilience {
  min-height: 60vh;
}

.resilicience__error-message {
  color: #FF0000;
  text-align: center;
  font-weight: bold;
  font-size: 20px;
  margin-top: 5em;
  margin-bottom: 1em;
}