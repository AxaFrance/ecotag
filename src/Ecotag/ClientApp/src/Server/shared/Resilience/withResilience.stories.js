import React from 'react';
import {resilienceStatus, withResilience} from '.';

export default {title: 'Resilience'};

const BasicComponent = () => <><p>Ceci est un composant basic de succès</p></>;

const ComponentWithResilience = withResilience(BasicComponent);

export const withError = () => <ComponentWithResilience status={resilienceStatus.ERROR}/>;

export const withForbidden = () => <ComponentWithResilience status={resilienceStatus.FORBIDDEN}/>;
