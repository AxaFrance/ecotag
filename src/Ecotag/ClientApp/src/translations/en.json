{
  "header": {
    "title": "An eco-system that aims to be eco-friendly",
    "errors": {
      "not_connected": "Not connected"
    }
  },
  "paging": {
    "display": "Display",
    "elements": "elements",
    "previous": "« Previous",
    "next": "Next »"
  },
  "stepper": {
    "confirm": "Confirm"
  },
  "resilience": {
    "error": {
      "first_part": "An error occurred.",
      "second_part": "We are aware of the problem. An investigation is ongoing..."
    },
    "forbidden": "You are not allowed to access this resource."
  },
  "not_found": {
    "title": "404 Page Not Found",
    "not": "not",
    "found": "found",
    "message": "The page you are looking for is not here !"
  },
  "home": {
    "title": "Home",
    "projects": "Projects",
    "datasets": "Datasets",
    "groups": "Teams",
    "errors": {
      "no_roles": "No role is associated to your profile."
    }
  },
  "group": {
    "title": "Teams",
    "subtitle": "Annotation teams governance",
    "creation_title": "Creation",
    "new": {
      "name_input": {
        "label": "Name : ",
        "help_message": "Ex : bank account details annotators"
      },
      "validation": "Validate"
    },
    "count_teams": {
      "first_part": "There is/are (",
      "second_part": ") team(s)"
    },
    "empty_list": "No element",
    "list": {
      "header": {
        "name": "Name",
        "users": "Users",
        "actions": "Actions"
      },
      "edit": "Edit",
      "edit_popup": {
        "header": "Add/Remove users from this team",
        "add_user_placeholder": "Add user",
        "footer": {
          "cancel": "Cancel",
          "validation": "Validate"
        }
      }
    }
  },
  "dataset": {
    "title": "Datasets",
    "subtitle": "A dataset is the representation of a collection of data",
    "new_dataset": "New dataset",
    "access_token": "Access token",
    "active_datasets_counter": "Your active datasets : ",
    "filter_by": "Filter by",
    "name": "Name",
    "empty_list": "No elements",
    "list": {
      "actions_titles": {
        "locked": "locked",
        "locked_and_work_in_progress": "locked and work in progress",
        "pending": "pending",
        "none": "unlocked"
      },
      "header": {
        "lock": "Lock",
        "name": "Name",
        "team": "Team",
        "classification": "Classification",
        "nb_files": "Number of files",
        "type": "Type",
        "creation_date": "Creation date",
        "action": "Action"
      },
      "edit": "Edit"
    },
    "new": {
      "title": "New dataset",
      "stepper": {
        "title": "New dataset"
      },
      "form_title": "New dataset",
      "name_input": {
        "label": "Name",
        "help_message": "Ex : bank-account-details-dataset"
      },
      "team_input": {
        "label": "Team"
      },
      "type_input": {
        "label": "Type",
        "help_message": "Ex : Image, Text",
        "options_labels": {
          "image": "Images (.jpg, .png, jpeg)",
          "document": "Documents (.txt, .eml, .tiff, .pdf, .doc, etc.)",
          "text": "Text (.txt)",
          "eml": "Emails (.eml)"
        },
        "help": {
          "title": "Information on the accepted documents types",
          "images": {
            "name": "Images",
            "types": " : .jpg, .png, jpeg"
          },
          "documents": {
            "name": "Documents",
            "types": " : .txt, .eml, .msg, .jpg, .png, jpeg, .tiff, .tif, .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .rtf, .odt, .ods, .odp, .zip"
          },
          "text": {
            "name": "Text",
            "types": " : .txt"
          },
          "eml": {
            "name": "Eml",
            "types": " : .eml"
          }
        }
      },
      "classification_input": {
        "label": "Classification",
        "help_message": "Ex : Critical",
        "options": {
          "public": {
            "label": "Public"
          },
          "internal": {
            "label": "Internal"
          },
          "confidential": {
            "label": "Confidential"
          },
          "critical": {
            "label": "Critical"
          }
        },
        "help": {
          "title": "Information on criticality",
          "public": {
            "name": "Public",
            "detail": " : a piece of information made deliberately for the public, or specifically made for public diffusion"
          },
          "internal": {
            "name": "Internal",
            "detail": " : a piece of information that is accessible to all or a large portion of employees and is not intended for external parties."
          },
          "confidential": {
            "name": "Confidential",
            "detail": " : a piece of information restricted to a specific audience"
          },
          "critical": {
            "name": "Critical",
            "detail": " : Please refer to your manager."
          }
        }
      },
      "import_input": {
        "label": "Import from a blob : "
      },
      "dataset_to_import": {
        "label": "Dataset to import : "
      },
      "validate": "Validation"
    },
    "confirm": {
      "title": "Confirmation",
      "stepper": "New dataset",
      "confirmation_title": "New dataset added !",
      "confirmation_description": "If you have imported a dataset, the process of adding files can take a while.",
      "return_button_label": "Go back to datasets list"
    },
    "edit": {
      "confirm_modal": {
        "title": "Do you want to lock the dataset permanently ?",
        "warning_first_part": "This action is permanent.",
        "warning_second_part": "Every modification (addition or deletion of files) will be impossible then."
      },
      "subtitle": "Dataset edition",
      "lock": "Lock",
      "locked_state": {
        "locked": "Dataset locked",
        "locked_and_work_in_progress": "Dataset is being worked on",
        "default": "Dataset is being imported"
      },
      "file_upload": {
        "title": "Upload files ",
        "file_input": {
          "label": "Browse",
          "placeholder": "Drag/drop your files here"
        },
        "send_button": {
          "label": "Send"
        },
        "errors": {
          "already_existing": {
            "first_part": "The file ",
            "second_part": " already exists"
          },
          "could_not_upload": {
            "first_part": "The file ",
            "second_part": " could not be uploaded"
          }
        }
      },
      "files_list": {
        "global_info_tab": {
          "title": "Global information",
          "name": "Name",
          "creation_date": "Creation date",
          "type": "Type",
          "nb_files": "Number of files",
          "files_total_size": "Total files size",
          "files_mean_size": "Average file size"
        },
        "files_list_tab": {
          "title": "Files list",
          "name": "Name",
          "type": "Type",
          "size": "Size",
          "action": "Action",
          "delete_action": "Delete"
        }
      }
    }
  }
}