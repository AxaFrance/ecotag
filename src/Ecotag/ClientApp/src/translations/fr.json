{
  "header": {
    "title": "Un écosystème qui vise à être écologique",
    "errors": {
      "not_connected": "Non connecté"
    }
  },
  "home": {
    "title": "Accueil",
    "projects": "Projets",
    "datasets": "Datasets",
    "groups": "Groupes",
    "errors": {
      "no_roles": "Aucun rôle n'est attribué à votre profil."
    }
  }
}