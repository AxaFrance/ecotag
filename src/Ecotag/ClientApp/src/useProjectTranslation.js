import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
