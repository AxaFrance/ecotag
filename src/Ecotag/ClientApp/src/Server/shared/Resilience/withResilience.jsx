import React from 'react';
import {resilienceStatus} from '.';
import Loader, {LoaderModes} from '@axa-fr/react-toolkit-loader';
import './withResilience.scss';
import useProjectTranslation from '../../../useProjectTranslation';

export const withResilience = Component => ({status, loaderText = null, ...otherProps}) => {
    const {translate} = useProjectTranslation();
    const {ERROR, SUCCESS, LOADING, POST, FORBIDDEN} = resilienceStatus;

    const setLoaderText = (loaderText, loaderMode) => {
        let result;
        if(loaderText !== null){
            return loaderText;
        }
        switch (loaderMode){
            case LoaderModes.get:
                result = translate('resilience.get');
                break;
            case LoaderModes.post:
                result = translate('resilience.post');
                break;
            default:
                result = null;
                break;
        }
        return result;
    }

    return (
        <>
            {
                {
                    [LOADING]: <Loader mode={LoaderModes.get} text={setLoaderText(loaderText, LoaderModes.get)}><Component
                        loaderMode={LoaderModes.get} {...otherProps} /></Loader>,
                    [POST]: <Loader mode={LoaderModes.get} text={setLoaderText(loaderText, LoaderModes.post)}> <Component
                        loaderMode={LoaderModes.post} {...otherProps} /></Loader>,
                    [ERROR]: (
                        <div className="resilience">
                            <div className="resilicience__error-message">
                                <p>{translate('resilience.error.first_part')}</p>
                                <br/>
                                <p>{translate('resilience.error.second_part')}</p>
                            </div>
                        </div>
                    ),
                    [FORBIDDEN]: (
                        <div className="resilience">
                            <div className="resilicience__error-message">
                                <p>{translate('resilience.forbidden')}</p>
                            </div>
                        </div>
                    ),
                    [SUCCESS]: <Component loaderMode={LoaderModes} {...otherProps} />
                }[status]
            }
        </>
    );
};
