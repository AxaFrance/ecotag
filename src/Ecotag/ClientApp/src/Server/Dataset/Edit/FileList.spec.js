import {fireEvent, render, waitFor} from '@testing-library/react';
import FileList from './FileList';
import {Locked} from '../Dataset.service';
import {resilienceStatus} from '../../shared/Resilience';
import {changeProjectTranslationLanguage} from '../../../translations/useProjectTranslation';

describe('FileList', () => {
    const givenState = {
        files: {
            filesSend: [{file: {size: 0, name: "testFile.png", type: 'Image', id: '0001'}}],
            filesSendError: [],
            filesLoad: [],
            filesLoadError: [],
            paging: {itemByPages: 50, currentPages: 1}
        },
        openLockModal: false,
        dataset: {
            id: '0001',
            name: 'datasetName',
            createDate: 0,
            type: 'Image',
            locked: Locked.None
        }
    };

    const renderComponentThenNavigateAndCheckSnapshots = async (language, expectedText) => {
        changeProjectTranslationLanguage(language);
        const {asFragment, getByText} = render(<FileList
            fetch={() => {}}
            state={givenState}
            setState={() => {}}/>);

        expect(asFragment()).toMatchSnapshot();
        const regex = new RegExp(expectedText, "i");
        const tabButton = getByText(regex);
        fireEvent.click(tabButton);
        await waitFor(() => expect(getByText(/testFile.png/i)).not.toBeNull());
        expect(asFragment()).toMatchSnapshot();
    };

    it('should render correctly in english', async () => {
        await renderComponentThenNavigateAndCheckSnapshots("en", "Files list");
    });
    it('should render correctly in french', async () => {
        await renderComponentThenNavigateAndCheckSnapshots("fr", "Liste des fichiers");
    })

    describe('deleteFile', () => {

        beforeAll(() => changeProjectTranslationLanguage('en'));

        const testCases = [
            [403, resilienceStatus.FORBIDDEN],
            [500, resilienceStatus.ERROR]
        ];

        it('should delete file correctly', async () => {
            const givenFetch = jest.fn()
                .mockResolvedValueOnce({status: 200});
            const givenSetState = jest.fn();
            const {getByText, getByTitle} = render(<FileList fetch={givenFetch} state={givenState}
                                                             setState={givenSetState}/>);
            const tabButton = getByText(/Files list/i);
            fireEvent.click(tabButton);
            await waitFor(() => expect(getByText(/testFile.png/i)).not.toBeNull());
            const deleteButton = getByTitle(/Delete/i);
            fireEvent.click(deleteButton);
            await waitFor(() => expect(givenSetState).toHaveBeenCalled());
            expect(givenSetState).toHaveBeenCalledWith({
                ...givenState,
                files: {...givenState.files, filesSend: []},
                status: resilienceStatus.SUCCESS
            });
        });
        test.each(testCases)(
            "given %p response, returns related resilience status",
            async (returnedStatus, expectedStatus) => {
                const givenFetch = jest.fn()
                    .mockResolvedValueOnce({status: returnedStatus});
                const givenSetState = jest.fn();
                const {getByText, getByTitle} = render(<FileList fetch={givenFetch} state={givenState}
                                                                 setState={givenSetState}/>);
                const tabButton = getByText(/Files list/i);
                fireEvent.click(tabButton);
                await waitFor(() => expect(getByText(/testFile.png/i)).not.toBeNull());
                const deleteButton = getByTitle(/Delete/i);
                fireEvent.click(deleteButton);
                await waitFor(() => expect(givenSetState).toHaveBeenCalled());
                expect(givenSetState).toHaveBeenCalledWith({...givenState, status: expectedStatus});
            }
        );
    });

    describe('downloadAsync', () => {

        beforeAll(() => changeProjectTranslationLanguage('en'));

        const testCases = [
            [403, resilienceStatus.FORBIDDEN],
            [500, resilienceStatus.ERROR]
        ];

        test.each(testCases)(
            "given %p response, returns related resilience status",
            async (returnedStatus, expectedStatus) => {
                const givenFetch = jest.fn()
                    .mockResolvedValueOnce({status: returnedStatus});
                const givenSetState = jest.fn();
                const {getByText} = render(<FileList fetch={givenFetch} state={givenState} setState={givenSetState}/>);
                const tabButton = getByText(/Files list/i);
                fireEvent.click(tabButton);
                await waitFor(() => expect(getByText(/testFile.png/i)).not.toBeNull());
                const downloadButton = getByText(/testFile.png/i);
                fireEvent.click(downloadButton);
                await waitFor(() => expect(givenSetState).toHaveBeenCalled());
                expect(givenSetState).toHaveBeenCalledWith({...givenState, status: expectedStatus});
            }
        );
    });
})