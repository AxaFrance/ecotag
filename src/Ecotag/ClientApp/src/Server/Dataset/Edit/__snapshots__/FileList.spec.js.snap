// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList should render correctly in english 1`] = `
<DocumentFragment>
  <div
    class="edit-dataset__file-list-container"
  >
    <div
      class="af-tabs af-tabs--container"
    >
      <ul
        class="af-tabs__control"
      >
        <li
          class="af-tabs__item af-tabs__item--active"
        >
          <button
            class="af-tabs__link"
            id="0"
          >
            General information
          </button>
        </li>
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="1"
          >
            Files list
          </button>
        </li>
      </ul>
      <div
        class="af-tabs__content"
      >
        <div
          class="af-tabs__pane"
        >
          <article
            class="af-restitution"
          >
            <section
              class="af-restitution__content"
            >
              <div
                class="col col-sm-12 col-md-12 col-lg-12 col-xl-12"
              >
                <div
                  class="row af-restitution__content-left"
                >
                  <div
                    class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
                  >
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Name
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        datasetName
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Creation date
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        01/01/0001
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Type
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        Image
                      </dd>
                    </dl>
                  </div>
                  <div
                    class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
                  >
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Number of files
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        1
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Total files size
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        0 Byte
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Average file size
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        0 Byte
                      </dd>
                    </dl>
                  </div>
                </div>
              </div>
            </section>
          </article>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FileList should render correctly in english 2`] = `
<DocumentFragment>
  <div
    class="edit-dataset__file-list-container"
  >
    <div
      class="af-tabs af-tabs--container"
    >
      <ul
        class="af-tabs__control"
      >
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="0"
          >
            General information
          </button>
        </li>
        <li
          class="af-tabs__item af-tabs__item--active"
        >
          <button
            class="af-tabs__link"
            id="1"
          >
            Files list
          </button>
        </li>
      </ul>
      <div
        class="af-tabs__content"
      >
        <div
          class="af-tabs__pane"
        >
          <table
            class="af-table"
          >
            <thead
              class="af-table-thead"
            >
              <tr
                class="af-table__tr"
              >
                <th
                  class="af-table__th"
                >
                  <span
                    class="af-table__th-content"
                  >
                    Name
                  </span>
                </th>
                <th
                  class="af-table__th"
                >
                  <span
                    class="af-table__th-content "
                  >
                    Type
                  </span>
                </th>
                <th
                  class="af-table__th"
                >
                  Size
                </th>
                <th
                  class="af-table__th"
                >
                  Action
                </th>
              </tr>
            </thead>
            <tbody
              class="af-table__body"
            >
              <tr
                class="af-table__tr"
              >
                <td
                  class="af-table__cell"
                >
                  <a
                    alt="testFile.png"
                    href="#testFile.png"
                  >
                    testFile.png
                  </a>
                </td>
                <td
                  class="af-table__cell"
                >
                  Image
                </td>
                <td
                  class="af-table__cell"
                >
                  0 Byte
                </td>
                <td
                  class="af-table__cell"
                >
                   
                  <a
                    class="btn af-btn--circle"
                    href="#"
                    id="deleteButton"
                    role="button"
                    tabindex="0"
                    title="Delete"
                  >
                    <i
                      class="glyphicon glyphicon-trash"
                    />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="af-paging"
          >
            <div
              class="af-paging__limit"
            >
              <div
                class="af-paging__limit"
              >
                <form
                  class="af-paging__form"
                >
                  <div
                    class="af-form__group"
                  >
                    <div
                      class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
                    >
                      <label
                        class="af-form__group-label"
                        for="home_paging"
                      >
                        Display
                      </label>
                    </div>
                    <div
                      class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
                    >
                      <div
                        class="af-form__select"
                      >
                        <div
                          class="af-form__select-container"
                        >
                          <select
                            class="af-form__input-select"
                            id="home_paging"
                          >
                            <option
                              value="5"
                            >
                              5
                            </option>
                            <option
                              value="10"
                            >
                              10
                            </option>
                            <option
                              value="25"
                            >
                              25
                            </option>
                            <option
                              value="50"
                            >
                              50
                            </option>
                            <option
                              value="100"
                            >
                              100
                            </option>
                          </select>
                          <span
                            class="glyphicon glyphicon-menu-down"
                          />
                        </div>
                        <span
                          class="af-form__input-cmplt"
                        >
                          elements
                        </span>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="af-paging__pager"
            >
              <nav
                class="af-pager"
              >
                <ul
                  class="af-pager__pagination"
                >
                  <li
                    class="af-pager__item af-pager__item--disabled"
                  >
                    <span
                      class="af-pager__item-nolink"
                    >
                      « Previous
                    </span>
                  </li>
                  <li
                    class="af-pager__item af-pager__item--active"
                  >
                    <a
                      class="af-pager__item-link"
                      href="#"
                    >
                      <span>
                        1
                      </span>
                    </a>
                  </li>
                  <li
                    class="af-pager__item af-pager__item--disabled"
                  >
                    <span
                      class="af-pager__item-nolink"
                    >
                      Next »
                    </span>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FileList should render correctly in french 1`] = `
<DocumentFragment>
  <div
    class="edit-dataset__file-list-container"
  >
    <div
      class="af-tabs af-tabs--container"
    >
      <ul
        class="af-tabs__control"
      >
        <li
          class="af-tabs__item af-tabs__item--active"
        >
          <button
            class="af-tabs__link"
            id="0"
          >
            Informations générales
          </button>
        </li>
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="1"
          >
            Liste des fichiers
          </button>
        </li>
      </ul>
      <div
        class="af-tabs__content"
      >
        <div
          class="af-tabs__pane"
        >
          <article
            class="af-restitution"
          >
            <section
              class="af-restitution__content"
            >
              <div
                class="col col-sm-12 col-md-12 col-lg-12 col-xl-12"
              >
                <div
                  class="row af-restitution__content-left"
                >
                  <div
                    class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
                  >
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Nom
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        datasetName
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Date de création
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        01/01/0001
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Type
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        Image
                      </dd>
                    </dl>
                  </div>
                  <div
                    class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
                  >
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Nombre de fichiers
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        1
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Poids total des fichiers
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        0 Byte
                      </dd>
                    </dl>
                    <dl
                      class="af-restitution__listdef"
                    >
                      <dt
                        class="af-restitution__listdef-item"
                      >
                        <span
                          class="af-restitution__text"
                        >
                          Poids moyen des fichiers
                        </span>
                      </dt>
                      <dd
                        class="af-restitution__listdef-value"
                      >
                        0 Byte
                      </dd>
                    </dl>
                  </div>
                </div>
              </div>
            </section>
          </article>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FileList should render correctly in french 2`] = `
<DocumentFragment>
  <div
    class="edit-dataset__file-list-container"
  >
    <div
      class="af-tabs af-tabs--container"
    >
      <ul
        class="af-tabs__control"
      >
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="0"
          >
            Informations générales
          </button>
        </li>
        <li
          class="af-tabs__item af-tabs__item--active"
        >
          <button
            class="af-tabs__link"
            id="1"
          >
            Liste des fichiers
          </button>
        </li>
      </ul>
      <div
        class="af-tabs__content"
      >
        <div
          class="af-tabs__pane"
        >
          <table
            class="af-table"
          >
            <thead
              class="af-table-thead"
            >
              <tr
                class="af-table__tr"
              >
                <th
                  class="af-table__th"
                >
                  <span
                    class="af-table__th-content"
                  >
                    Nom
                  </span>
                </th>
                <th
                  class="af-table__th"
                >
                  <span
                    class="af-table__th-content "
                  >
                    Type
                  </span>
                </th>
                <th
                  class="af-table__th"
                >
                  Poids
                </th>
                <th
                  class="af-table__th"
                >
                  Action
                </th>
              </tr>
            </thead>
            <tbody
              class="af-table__body"
            >
              <tr
                class="af-table__tr"
              >
                <td
                  class="af-table__cell"
                >
                  <a
                    alt="testFile.png"
                    href="#testFile.png"
                  >
                    testFile.png
                  </a>
                </td>
                <td
                  class="af-table__cell"
                >
                  Image
                </td>
                <td
                  class="af-table__cell"
                >
                  0 Byte
                </td>
                <td
                  class="af-table__cell"
                >
                   
                  <a
                    class="btn af-btn--circle"
                    href="#"
                    id="deleteButton"
                    role="button"
                    tabindex="0"
                    title="Supprimer"
                  >
                    <i
                      class="glyphicon glyphicon-trash"
                    />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="af-paging"
          >
            <div
              class="af-paging__limit"
            >
              <div
                class="af-paging__limit"
              >
                <form
                  class="af-paging__form"
                >
                  <div
                    class="af-form__group"
                  >
                    <div
                      class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
                    >
                      <label
                        class="af-form__group-label"
                        for="home_paging"
                      >
                        Afficher
                      </label>
                    </div>
                    <div
                      class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
                    >
                      <div
                        class="af-form__select"
                      >
                        <div
                          class="af-form__select-container"
                        >
                          <select
                            class="af-form__input-select"
                            id="home_paging"
                          >
                            <option
                              value="5"
                            >
                              5
                            </option>
                            <option
                              value="10"
                            >
                              10
                            </option>
                            <option
                              value="25"
                            >
                              25
                            </option>
                            <option
                              value="50"
                            >
                              50
                            </option>
                            <option
                              value="100"
                            >
                              100
                            </option>
                          </select>
                          <span
                            class="glyphicon glyphicon-menu-down"
                          />
                        </div>
                        <span
                          class="af-form__input-cmplt"
                        >
                          éléments
                        </span>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="af-paging__pager"
            >
              <nav
                class="af-pager"
              >
                <ul
                  class="af-pager__pagination"
                >
                  <li
                    class="af-pager__item af-pager__item--disabled"
                  >
                    <span
                      class="af-pager__item-nolink"
                    >
                      « Précédent
                    </span>
                  </li>
                  <li
                    class="af-pager__item af-pager__item--active"
                  >
                    <a
                      class="af-pager__item-link"
                      href="#"
                    >
                      <span>
                        1
                      </span>
                    </a>
                  </li>
                  <li
                    class="af-pager__item af-pager__item--disabled"
                  >
                    <span
                      class="af-pager__item-nolink"
                    >
                      Suivant »
                    </span>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
